app/services/bulk_upload/lettings/year2023/csv_parser.rb

```diff
@@ -61,6 +63,12 @@ def too_many_columns?
     max_columns_count > MAX_COLUMNS
   end
 
+  def correct_template_for_year?
+    collection_start_year_for_date(first_record_start_date) == 2023
+  rescue Date::Error
+    true
```

Comment:
Would the correct rescue not return false here, as in, we couldn't construct a valid start date to determine validity?

Reply:
So, the start date for 2023 would need to be constructed from different fields. This is using the field numbers for 22/23 and checking if a 22/23 date can be constructed and if yes, then we know for sure that this is a template for 22/23. Otherwise we default to saying that the template is correct, cause there could be errors, for example the user forgot to fill in a part of the date etc

Reply:
Ah, this method is a bit more lenient is that it assumes the template is correct if anything goes wrong.

How about calling it `assumed_correct_template_for_year?`?

Reply:
How about now, I reversed it and changed to wrong_template_for_year?